src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearingpart/AmendScheduledListingAction.java

```diff
@@ -0,0 +1,109 @@
+package uk.gov.hmcts.reform.sandl.snlevents.actions.hearingpart;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.val;
+import uk.gov.hmcts.reform.sandl.snlevents.actions.Action;
+import uk.gov.hmcts.reform.sandl.snlevents.actions.interfaces.RulesProcessable;
+import uk.gov.hmcts.reform.sandl.snlevents.exceptions.SnlEventsException;
+import uk.gov.hmcts.reform.sandl.snlevents.messages.FactMessage;
+import uk.gov.hmcts.reform.sandl.snlevents.model.Status;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.HearingPart;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.UserTransactionData;
+import uk.gov.hmcts.reform.sandl.snlevents.model.request.AmendScheduledListing;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.db.HearingPartRepository;
+import uk.gov.hmcts.reform.sandl.snlevents.service.RulesService;
+
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoField;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import javax.persistence.EntityManager;
+
+public class AmendScheduledListingAction extends Action implements RulesProcessable {
+
+    protected AmendScheduledListing amendScheduledListing;
+    protected HearingPart hearingPart;
+    protected String previousHearingPart;
+    protected String previousHearing;
+
+    protected HearingPartRepository hearingPartRepository;
+    protected EntityManager entityManager;
+
+    @SuppressWarnings("squid:S00107") // we intentionally go around DI here as such the amount of parameters
+    public AmendScheduledListingAction(AmendScheduledListing amendScheduledListing,
+                                       HearingPartRepository hearingPartRepository,
+                                       EntityManager entityManager,
+                                       ObjectMapper objectMapper) {
+        this.amendScheduledListing = amendScheduledListing;
+        this.hearingPartRepository = hearingPartRepository;
+        this.entityManager = entityManager;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public void getAndValidateEntities() {
+        hearingPart = hearingPartRepository.findOne(amendScheduledListing.getHearingPartId());
+
+        if (hearingPart == null) {
+            throw new SnlEventsException("Hearing part cannot be null!");
+        }
+
+        if (!hearingPart.getStatus().getStatus().equals(Status.Listed)) {
```

Comment:
Question:
Should we move logic for each "action" to separate class as we did for status change action?
if yes, we could do sth like this:
```
      if (!statusServiceManager.canBeUnlisted(hearing)) {
            throw new SnlEventsException("Hearing can not be unlisted");
        }
```

Reply:
How would this method look like then? I'm afraid it would become a one liner calling just that method?

public void getAndValidateEntities() {
   amendScheduledListingActionValidator.validate(obj)
}